src/components/map/layers/RoutePathLinkLayer.tsx

```diff
@@ -85,14 +95,33 @@ class RoutePathLinkLayer extends Component<RoutePathLinkLayerProps> {
         );
     }
 
+    private renderDirectionDecoration() {
+        const routePathLinks = this.props.routePathLinks;
+
+        const geoms = routePathLinks
+            .map(routePathLink => routePathLink.geometry);
+
+        return createCoherentLinesFromPolylines(geoms)
+            .map((geom, index) => (
+                <ArrowDecorator
+                    key={index}
+                    color={this.props.color}
+                    geometry={geom}
+                    onClick={this.props.onClick(this.layerRef)}
+                />
+            ),
```

Comment:
What createCoherentLinesFromPolylines does and why? I guess there must be an easier way without that helper function

Much clearner way would be to say:
```
return this.props.routePathLinks.map(routePathLink) =>
    routePathLink.geometry.map(polyline, index) => (
                <ArrowDecorator
                    key={index}
                    color={this.props.color}
                    geometry={geom}
                    onClick={this.props.onClick(this.layerRef)}
                />
           )
```
Would code like this work?

Reply:
It would work, but this creates arrow decoration from a set of polylines, not from one complete polyline. Having decorations made from a set of polylines makes the distance between the arrows inconsistent. Having one single polyline makes the arrows be with consistent distance between each other, throughout the whole coherent polyline.

Reply:
Add comment above the method?
/**
    Creates coherent polylines from multiple polylines: if there is a hole inside a polyline, it splits the polyline into two polylines etc.
*/
const createCoherentLinesFromPolylines = ...

Reply:
I need to change the comment a bit, the polylines don't have holes. There might be holes between the polylines, and this is where we should split the final polyline.